lib/gettext/plural.ex

```diff
@@ -122,18 +122,51 @@ defmodule Gettext.Plural do
 
   """
 
+  alias Expo.Messages
+
+  # Types
+
+  @type locale :: String.t()
+
+  @type pluralization_context :: %{
+          optional(:plural_forms_header) => String.t(),
+          required(:locale) => locale()
+        }
+
+  @type opts :: locale() | term()
+
   # Behaviour definition.
 
+  @doc """
+  Initialize context for `nplurals/1` / `plurals/2`.
+
+  Perform all preparations (for example parsing the plural forms header) per
+  for the provided locale to offload into compile time.
+
+  If the `init/1` callback is not defined, the `opts` will be set to the
+  `locale`.
+  """
+  @callback init(pluralization_context()) :: opts()
+
   @doc """
   Returns the number of possible plural forms in the given `locale`.
   """
-  @callback nplurals(locale :: String.t()) :: pos_integer
+  @callback nplurals(opts()) :: pos_integer()
 
   @doc """
   Returns the plural form in the given `locale` for the given `count` of
   elements.
   """
-  @callback plural(locale :: String.t(), count :: integer) :: plural_form :: non_neg_integer
+  @callback plural(opts(), count :: integer()) :: non_neg_integer()
```

Comment:
Nit: let's leave the name, it's helpful in documentation.

```suggestion
  @callback plural(opts(), count :: integer()) :: plural_form :: non_neg_integer()
```